README.md

```diff
@@ -96,6 +96,61 @@ If you are interested in helping, please reach out!
 You can use GitHub or our [Discord server](https://discord.gg/pxrBmy4).
 
 
+## Android Integration
+
+The Android implementation requires the application's Activity to forward input events (and
+input devices) to the plugin. Below is an example of a MainActivity for a clean Flutter project
+that has implemented the required boilerplate code. For many projects it will be possible to simply
+duplicate this setup.
+
+```dart
+package [YOUR_PACKAGE_NAME]
+
+import android.hardware.input.InputManager
+import android.os.Handler
+import android.view.InputDevice
+import android.view.KeyEvent
+import android.view.MotionEvent
+import io.flutter.embedding.android.FlutterActivity
+import dev.markvideon.gamepads_android.GamepadsCompatibleActivity
+
+class MainActivity: FlutterActivity(), GamepadsCompatibleActivity {
+    var keyListener: ((KeyEvent) -> Boolean)? = null
+    var motionListener: ((MotionEvent) -> Boolean)? = null
+
+    override fun dispatchGenericMotionEvent(motionEvent: MotionEvent): Boolean {
+        motionListener?.invoke(motionEvent)
+        return true
+    }
+
+    override fun dispatchKeyEvent(keyEvent: KeyEvent): Boolean {
+        keyListener?.invoke(keyEvent)
+        return true
+    }
+
+    override fun isGamepadsInputDevice(device: InputDevice): Boolean {
+        return device.sources and InputDevice.SOURCE_GAMEPAD == InputDevice.SOURCE_GAMEPAD
+            || device.sources and InputDevice.SOURCE_JOYSTICK == InputDevice.SOURCE_JOYSTICK
+    }
```

Comment:
why would users need to override this method from GamepadsCompatibleActivity?
can't we just provide the impl for this?

I would prefer to minimize any code that users have to add manually to use the libary

Reply:
on a similar line of thought, I wonder if we can provide a GamepadsCompatibleActivityAdapter, that implements all these required methods on behalf of the user just requiring the minimum hooks to be added:
* wire in registerKeyEventHandler and registerMotionEventHandler
* access to getSystemService(INPUT_SERVICE)
it appears that everything else is boilerplate that would be the same for all (or most) users using these 3 variables and thus could be hidden away by us
but I might be missing something, I am only looking at the code on github. I can try to pull on IJ later to take a look